eval/src/tests/instruction/mixed_simple_join_function/mixed_simple_join_function_test.cpp

```diff
@@ -41,78 +41,95 @@ std::ostream &operator<<(std::ostream &os, Overlap overlap)
-        .add("a", GenSpec(1.5))
-        .add("b", GenSpec(2.5))
-        .add("sparse", GenSpec().map("x", {"a", "b", "c"}))
-        .add("mixed", GenSpec().map("x", {"a", "b", "c"}).idx("y", 5).idx("z", 3))
-        .add("empty_mixed", GenSpec().map("x", {}).idx("y", 5).idx("z", 3))
-        .add_mutable("@mixed", GenSpec().map("x", {"a", "b", "c"}).idx("y", 5).idx("z", 3))
-        .add_variants("a1b1c1", GenSpec().idx("a", 1).idx("b", 1).idx("c", 1))
-        .add_variants("x1y1z1", GenSpec().idx("x", 1).idx("y", 1).idx("z", 1))
-        .add_variants("x3y5z3", GenSpec().idx("x", 3).idx("y", 5).idx("z", 3))
-        .add_variants("z3", GenSpec().idx("z", 3))
-        .add_variants("c5d1", GenSpec().idx("c", 5).idx("d", 1))
-        .add_variants("b1c5", GenSpec().idx("b", 1).idx("c", 5))
-        .add_variants("x3y5", GenSpec().idx("x", 3).idx("y", 5).seq([](size_t idx) noexcept { return double((idx * 2) + 3); }))
-        .add_variants("x3y5$2", GenSpec().idx("x", 3).idx("y", 5).seq([](size_t idx) noexcept { return double((idx * 3) + 2); }))
-        .add_variants("y5", GenSpec().idx("y", 5).seq([](size_t idx) noexcept { return double((idx * 2) + 3); }))
-        .add_variants("y5$2", GenSpec().idx("y", 5).seq([](size_t idx) noexcept { return double((idx * 3) + 2); }))
-        .add_variants("y5z3", GenSpec().idx("y", 5).idx("z", 3).seq([](size_t idx) noexcept { return double((idx * 2) + 3); }))
-        .add_variants("y5z3$2", GenSpec().idx("y", 5).idx("z", 3).seq([](size_t idx) noexcept { return double((idx * 3) + 2); }));
-}
-EvalFixture::ParamRepo param_repo = make_params();
-
-void verify_optimized(const vespalib::string &expr, Primary primary, Overlap overlap, bool pri_mut, size_t factor, int p_inplace = -1) {
-    EvalFixture slow_fixture(prod_factory, expr, param_repo, false);
-    EvalFixture fixture(prod_factory, expr, param_repo, true, true);
-    EXPECT_EQUAL(fixture.result(), EvalFixture::ref(expr, param_repo));
-    EXPECT_EQUAL(fixture.result(), slow_fixture.result());
-    auto info = fixture.find_all<MixedSimpleJoinFunction>();
-    ASSERT_EQUAL(info.size(), 1u);
-    EXPECT_TRUE(info[0]->result_is_mutable());
-    EXPECT_EQUAL(info[0]->primary(), primary);
-    EXPECT_EQUAL(info[0]->overlap(), overlap);
-    EXPECT_EQUAL(info[0]->primary_is_mutable(), pri_mut);
-    EXPECT_EQUAL(info[0]->factor(), factor);
-    EXPECT_TRUE((p_inplace == -1) || (fixture.num_params() > size_t(p_inplace)));
-    for (size_t i = 0; i < fixture.num_params(); ++i) {
-        if (i == size_t(p_inplace)) {
-            EXPECT_EQUAL(fixture.get_param(i), fixture.result());
-        } else {
-            if (!fixture.result().cells().empty()) {
-                EXPECT_NOT_EQUAL(fixture.get_param(i), fixture.result());
+struct FunInfo {
+    using LookFor = MixedSimpleJoinFunction;
+    Overlap overlap;
+    size_t factor;
+    std::optional<Primary> primary;
+    bool l_mut;
+    bool r_mut;
+    bool require_inplace;
```

Comment:
should probably be optional to model potential knowledge of either true or false